frontend/src/metabase-lib/v1/parameters/utils/targets.unit.spec.ts

```diff
@@ -133,4 +244,336 @@ describe("parameters/utils/targets", () => {
+        });
+
+        it("date breakouts in multiple stages - returns date column from the last stage only", () => {
+          const question = createModel({
+            dataset_query: queryDateBreakoutsMultiStage,
+            result_metadata: [createOrdersCreatedAtField(), createCountField()],
+          });
+          const { columns } = getParameterColumns(question, parameter);
+
+          expect(columns).toHaveLength(0);
+        });
+      });
+    });
+
+    describe("date parameter", () => {
+      const parameter = createDateParameter();
+
+      describe("question", () => {
+        it("returns date columns from source table and implicitly joinable tables", () => {
+          const question = createQuestion();
+          const { query, stageIndex, columns } = getParameterColumns(
+            question,
+            parameter,
+          );
+          const columnsInfos = getColumnsInfos(query, stageIndex, columns);
+
+          expect(columnsInfos).toEqual([
+            ["Orders", "Created At"],
+            ["Products", "Product → Created At"],
+            ["People", "User → Birth Date"],
+            ["People", "User → Created At"],
+          ]);
+        });
+      });
+
+      describe("model", () => {
+        it("returns date columns from source table and implicitly joinable tables", () => {
+          const question = createModel({
+            result_metadata: createOrdersTable().fields,
+          });
+          const { query, stageIndex, columns } = getParameterColumns(
+            question,
+            parameter,
+          );
+          const columnsInfos = getColumnsInfos(query, stageIndex, columns);
+
+          expect(columnsInfos).toEqual([
+            ["Question", "Created At"],
+            ["People", "User → Birth Date"],
+            ["People", "User → Created At"],
+            ["Products", "Product → Created At"],
+          ]);
+        });
+      });
+    });
+  });
 });
+
+function createUnitOfTimeParameter() {
+  return createMockParameter({
+    name: "Unit of Time",
+    slug: "unit_of_time",
+    id: "49358513",
+    type: "temporal-unit",
+    sectionId: "temporal-unit",
+  });
+}
+
+function createDateParameter() {
+  return createMockParameter({
+    name: "Date",
+    slug: "date",
+    id: "57ab6554",
+    type: "date/all-options",
+    sectionId: "date",
+  });
+}
+
+function createQuestion(opts?: Partial<StructuredCard>) {
+  const card = createSavedStructuredCard(opts);
+
+  return new Question(card, metadata);
+}
+
+function createCountField() {
+  return createMockField({
+    display_name: "Count",
+    semantic_type: "type/Quantity",
+    field_ref: ["aggregation", 0],
+    base_type: "type/BigInteger",
+    effective_type: "type/BigInteger",
+    name: "count",
+  });
+}
+
+function createModel(opts: Partial<StructuredCard>) {
+  if (!("result_metadata" in opts)) {
+    throw new Error(
+      "result_metadata needs to be passed, otherwise test setup would be incorrect",
+    );
+  }
```

Comment:
It took me a while to figure out why one of the model tests was failing, so I'm leaving this here.
It can cause false positives too.